Sources/PointFree/Account/Cancel.swift

```diff
@@ -0,0 +1,192 @@
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T3<Database.Subscription, Database.User, A>, Data>
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, T3<Database.Subscription, Database.User, A>, Data> {
+
+    return { conn in
+      guard conn.data.first.userId == conn.data.second.first.id else {
+        return conn
+          |> writeStatus(.notFound)
+          >-> respond(text: "Not the subscription owner :(")
+      }
+
+      return conn |> middleware
+    }
+}
+
+func requireStripeSubscription<A>(_ suchThat: @escaping (Stripe.Subscription) -> Bool)
+  -> (@escaping Middleware<StatusLineOpen, ResponseEnded, T3<Stripe.Subscription, Database.User, A>, Data>)
+  -> Middleware<StatusLineOpen, ResponseEnded, T3<Database.Subscription, Database.User, A>, Data> {
+
+    return { middleware in
+      { conn in
+
+        let (databaseSub, user, rest) = (conn.data.first, conn.data.second.first, conn.data.second.second)
+
+        return AppEnvironment.current.stripe.fetchSubscription(databaseSub.stripeSubscriptionId)
+          .run
+          .map(^\.right)
+          .flatMap { sub in
+
+            guard let sub = sub, suchThat(sub) else {
+              return conn
+                |> writeStatus(.notFound)
+                >-> respond(text: "Stripe subscription not found :(")
+            }
+
+            return conn.map(const(.init(first: sub, second: .init(first: user, second: rest))))
+              |> middleware
+        }
+      }
+    }
+}
+
+let confirmCancelView = View<(Stripe.Subscription, Database.User)> { subscription, currentUser in
+  document([
+    html([
+      head([
+        style(renderedNormalizeCss),
+        style(styleguide),
+        style(render(config: pretty, css: pricingExtraStyles)),
+        meta(viewport: .width(.deviceWidth), .initialScale(1)),
+        ]),
+      body(
+        darkNavView.view((currentUser, nil))
+          <> [
+            gridRow([
+              gridColumn(sizes: [.mobile: 12, .desktop: 8], [style(margin(leftRight: .auto))], [
+                div(
+                  [`class`([Class.padding([.mobile: [.all: 3], .desktop: [.all: 4]])])],
+                  titleRowView.view(unit)
+                    <> formRowView.view(unit)
+                )
+              ])
+          ]
+          <> footerView.view(unit)
+      )
+      ])
+    ])
+  ])
+}
+
+private let titleRowView = View<Prelude.Unit> { _ in
+  gridRow([`class`([Class.padding([.mobile: [.bottom: 4]])])], [
+    gridColumn(sizes: [.mobile: 12], [
+      div([
+        h1([`class`([Class.pf.type.title2])], ["Cancel Subscription?"])
+        ])
+      ])
+    ])
+}
+
+private let formRowView = View<Prelude.Unit> { _ in
+  gridRow([`class`([Class.padding([.mobile: [.bottom: 4]])])], [
+    gridColumn(sizes: [.mobile: 12], [
+      form([action(path(to: .cancel)), method(.post)], [
+        button(
+          [`class`([Class.pf.components.button(color: .red), Class.margin([.mobile: [.top: 3]])])],
```

Comment:
was thinking we could just use an underline style for this button instead of a full on button, to not make it stand out and seem too enticing.

we could add a `style` param to buttons so that  we could do `button(color: .purple, style: .filled)` vs `button(color: .red, style: .underlined)` vs `button(color: .red, style: .outlined)`.

Reply:
sorry this button is good, I meant for the first cancel button.